src/wasm/extract.cpp

```diff
@@ -512,7 +520,14 @@ std::string extractor::extract(const std::string& list_json) {
   log_info << "Done. Creating ZIP file.";
   save_zip();
 
-  return json::object().dump();
+  ret["status"]="Completed successfully";
+  return ret.dump();
+
+  ret_abort:
```

Comment:
Since this is only reachable via `goto`, why not just move that code to the place, where `goto` is, i.e. `if (aborted)`? `return` will end the loop anyway.